core/src/klio_core/config/core.py

```diff
@@ -83,6 +84,24 @@ class KlioConfig(BaseKlioConfig):
         job_config (KlioJobConfig): Job-related configuration.
     """
 
+    def __init__(
+        self,
+        raw_config_data,
+        raw_overrides=[],
+        raw_templates=[],
```

Comment:
For these two args, I'd suggest setting them to `None`, then in the actual function, set them to an empty list if they're equal to `None`. Setting defaults to something mutable [can cause unexpected behavior](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument), so unless there's a clear reason why we can't avoid it - even if it's not possible to encounter this unexpected behavior - I'd prefer to avoid it entirely.